identity-apps-core/apps/authentication-portal/src/main/webapp/emailOtp.jsp

```diff
@@ -79,10 +79,12 @@
         authenticationFailed = "true";
 
         if (request.getParameter(Constants.AUTH_FAILURE_MSG) != null) {
-            errorMessage = request.getParameter(Constants.AUTH_FAILURE_MSG);
+            String error = request.getParameter(Constants.AUTH_FAILURE_MSG);
 
-            if (errorMessage.equalsIgnoreCase("authentication.fail.message")) {
+            if (error.equalsIgnoreCase("authentication.fail.message")) {
                 errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, "error.retry.code.invalid");
+            } else if (!error.equalsIgnoreCase(AuthenticationEndpointUtil.i18n(resourceBundle, error))) {
+                errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, error);
             }
```

Comment:
When matching i18 key found.

<img width="1025" height="740" alt="Image" src="https://github.com/user-attachments/assets/4955cea3-299b-42c2-a5a1-8c63037a8f3e" />

When matching i18 key not found.


<img width="531" height="615" alt="Image" src="https://github.com/user-attachments/assets/9a8671ee-382a-4e53-9272-e3300b447111" />